api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpRecencyHTSClientCardDataExportReport2019.java

```diff
@@ -0,0 +1,188 @@
+
+	@Override
+	public String getDescription() {
+		return "Data Export for 2019 Version of the Recency HTS Client Card" ;
+	}
+
+	@Override
+	public String getName() {
+		return "Recency HTS Client Card Data Export 2019";
+	}
+
+	@Override
+	public String getUuid() {
+		return "fca24a48-c72d-4508-ad29-5c1fbd2271c1";
+	}
+
+	@Override
+	public String getVersion() {
+		return "1.1.0";
+	}
+
+	/**
+	 * @return the uuid for the report design for exporting to Excel
+	 */
+	@Override
+	public String getExcelDesignUuid() {
+		return "dc766b50-eb86-425a-b42e-cb5fc3f3a76d";
+	}
+
+	@Override
+	public List<Parameter> getParameters() {
+		List<Parameter> l = new ArrayList<Parameter>();
+		l.add(df.getStartDateParameter());
+		l.add(df.getEndDateParameter());
+		return l;
+	}
+
+	@Override
+	public List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+		List<ReportDesign> l = new ArrayList<ReportDesign>();
+		l.add(buildReportDesign(reportDefinition));
+		return l;
+	}
+
+	/**
+	 * Build the report design for the specified report, this allows a user to override the report
+	 * design by adding properties and other metadata to the report design
+	 *
+	 * @param reportDefinition
+	 * @return The report design
+	 */
+	@Override
+	public ReportDesign buildReportDesign(ReportDefinition reportDefinition) {
+		ReportDesign rd = createCSVDesign(getExcelDesignUuid(), reportDefinition);
+		return rd;
+	}
+
+
+	@Override
+	public ReportDefinition constructReportDefinition() {
+
+		ReportDefinition rd = new ReportDefinition();
+		rd.setUuid(getUuid());
+		rd.setName(getName());
+		rd.setDescription(getDescription());
+		rd.setParameters(getParameters());
+		rd.addDataSetDefinition("HTSCARDDATAEXPORT", Mapped.mapStraightThrough(dataSetDefinition()));
+		return rd;
+	}
+	private Concept getConcept(String uuid) {
+		return Dictionary.getConcept(uuid);
+	}
+
+	private CohortDefinition getPatientWithHCTEncounterDuringPeriod(){
+		return df.getAnyEncounterOfTypesBetweenDates(
+				Arrays.asList(Dictionary.getEncounterType("264daIZd-f80e-48fe-nba9-P37f2W1905Pv")));
+	}
+	private DataSetDefinition dataSetDefinition() {
+		PatientDataSetDefinition dsd = new PatientDataSetDefinition();
+		dsd.setName("HTS");
+		dsd.addParameters(getParameters());
+		dsd.addRowFilter(Mapped.mapStraightThrough(getPatientWithHCTEncounterDuringPeriod()));
+
+		//start constructing of the dataset
+		PersonAttributeType maritalStatus = Context.getPersonService().getPersonAttributeTypeByUuid("dce0c134-30ab-102d-86b0-7a5022ba4115");
+		PatientIdentifierType NIN = MetadataUtils.existing(PatientIdentifierType.class,"f0c16a6d-dc5f-4118-a803-616d0075d282");
+
+		//identifier
+		DataConverter identifierFormatter = new ObjectFormatter("{identifier}");
+
+
+		dsd.addColumn("serial_number", sdd.definition("serialNo",  getConcept("1646AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
```

Comment:
@echiteri Add the uuid for the encounter so that you can have a unique key for the encounter

Reply:
@ssmusoke I am not clear on what datadefinition library to use to get the encounter_uuid? I have the column but empty. We agreed that we don't need it for now